android/src/main/java/bbct/android/common/activity/MainActivity.java

```diff
@@ -189,4 +191,17 @@ public boolean onOptionsItemSelected(MenuItem item) {
 
         return super.onOptionsItemSelected(item);
     }
+
+    @Override
+    public void onBackPressed() {
+        int count = getFragmentManager().getBackStackEntryCount();
+
+        if (count == 0) {
+            super.onBackPressed();
+            //additional code
+        } else {
+            getFragmentManager().popBackStack();
+        }
+
+    }
```

Comment:
Is this implementation necessary? Does the default implementation in Activity already do this?